app/uk/gov/hmrc/transitmovementsrouter/connectors/PersistenceConnector.scala

```diff
@@ -77,8 +78,11 @@ class PersistenceConnectorImpl @Inject() (httpClientV2: HttpClientV2, appConfig:
         .withBody(source)
         .execute[Either[UpstreamErrorResponse, HttpResponse]]
         .map {
-          case Right(_) =>
-            Right(())
+          case Right(res) =>
+            Json
+              .fromJson[MessageId](Json.parse(res.body))
```

Comment:
...wait, does this work?

The body you're expecting is 

```
{
    "messageId": "ksjdfhgkjhsdgb"
}
```

`MessageId` is just a value, that would just try to parse a `JsString`...

Reply:
Oh, I see what you've done later on...